File: src/main/java/com/range/rpms/users/service/impl/UserServiceImpl.java
package com.range.rpms.users.service.impl;

import com.range.rpms.users.domain.model.User;
import com.range.rpms.users.domain.repository.UserRepository;
import com.range.rpms.users.exception.UserNotFoundException;
import com.range.rpms.users.service.UserService;
import org.springframework.stereotype.Service;

@Service
public class UserServiceImpl implements UserService {
    private final UserRepository userRepository;
    public UserServiceImpl(UserRepository userRepository) {
        this.userRepository = userRepository;
    }



    @Override
    public User findUser(Long id) {
     return userRepository.findById(id).orElseThrow((()->
             new UserNotFoundException("User not found")));
    }

    @Override
    public User findUserByUsername(String username) {
        return userRepository.findByUsername(username).orElseThrow(()->
                new UserNotFoundException("User not found"));
    }

}

File: src/main/kotlin/com/range/rpms/users/service/impl/UserServiceImpl.kt
package com.range.rpms.users.service.impl

import com.range.rpms.users.domain.entity.User
import com.range.rpms.users.domain.repository.UserRepository
import com.range.rpms.users.service.UserService
import org.springframework.stereotype.Service

@Service
class UserServiceImpl(
    private val userRepository: UserRepository,
) : UserService {
    override fun findByUsername(username: String): User? {
        return userRepository.findByUsername(username)

    }

    override fun findByEmail(email: String): User? {
        return userRepository.findByEmail(email)

    }

    override fun findByEmailORUsername(emailORUsername: String): User? {
        return findByEmail(emailORUsername)
            ?: findByUsername(emailORUsername)

    }

    override fun save(user: User): User {
        return userRepository.save(user)
    }


}